helix-core/src/main/java/org/apache/helix/controller/dataproviders/ResourceControllerDataProvider.java

```diff
@@ -573,11 +575,16 @@ public WagedInstanceCapacity getWagedInstanceCapacity() {
     return _wagedInstanceCapacity;
   }
 
-  private void buildSimpleCapacityMap(int globalMaxPartitionAllowedPerInstance) {
+  private void buildSimpleCapacityMap() {
```

Comment:
> If this is here because StickyRebalanceStrategy relies on global node capacity

Yeah, this is exact the reason - we need to populate the node usage globally before the rebalance stage. `StatefulRebalancer` seems for rebalancer, but here we need the rebalance strategy. To me the data provider is the glue for rebalancer and rebalance strategy.